tests/main/interfaces-fuse_support/task.yaml

```diff
@@ -34,14 +34,15 @@ prepare: |
     mkdir -p "$MOUNT_POINT_OUTSIDE"
 
 restore: |
-    # remove the mount point
-    mounts=$(nsenter "--mount=/run/snapd/ns/$NAME.mnt" cat /proc/mounts | \
-             grep "$(basename "$MOUNT_POINT") fuse" | cut -f2 -d' ')
-    for m in $mounts; do
-        nsenter "--mount=/run/snapd/ns/$NAME.mnt" umount "$m"
-    done
+    # Unmount the filesystem mounted inside the mount namespace of the
+    # application. Due to propagation settings this is sufficient even if
+    # instances are used, where the mount point was visible under two
+    # locations, because the unmount event propagates across.
```

Comment:
This requires some more explaining. The mount propagation is actually shared here. I will iterate and explain tomorrow.

Reply:
Oh right, because it was done earlier and not undone later:
```
mount("/etc", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_BIND|MS_REC, NULL) = 0   
mount("none", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_REC|MS_SLAVE, NULL) = 0   
mount("none", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_REC|MS_SHARED, NULL) = 0
...
mount("/snap/core18/current/etc/ssl", "/tmp/snap.rootfs_5fTIlC/etc/ssl", NULL, MS_BIND, NULL) = 0
mount("none", "/tmp/snap.rootfs_5fTIlC/etc/ssl", NULL, MS_SLAVE, NULL) = 0      
mount("/snap/core18/current/etc/nsswitch.conf", "/tmp/snap.rootfs_5fTIlC/etc/nsswitch.conf", NULL, MS_BIND, NULL) = 0
mount("none", "/tmp/snap.rootfs_5fTIlC/etc/nsswitch.conf", NULL, MS_SLAVE, NULL) = 0
```

Reply:
> Oh right, because it was done earlier and not undone in between the calls:
> 
> ```
> mount("/etc", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_BIND|MS_REC, NULL) = 0   
> mount("none", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_REC|MS_SLAVE, NULL) = 0   
> mount("none", "/tmp/snap.rootfs_5fTIlC//etc", NULL, MS_REC|MS_SHARED, NULL) = 0
> ...
> mount("/snap/core18/current/etc/ssl", "/tmp/snap.rootfs_5fTIlC/etc/ssl", NULL, MS_BIND, NULL) = 0
> mount("none", "/tmp/snap.rootfs_5fTIlC/etc/ssl", NULL, MS_SLAVE, NULL) = 0      
> mount("/snap/core18/current/etc/nsswitch.conf", "/tmp/snap.rootfs_5fTIlC/etc/nsswitch.conf", NULL, MS_BIND, NULL) = 0
> mount("none", "/tmp/snap.rootfs_5fTIlC/etc/nsswitch.conf", NULL, MS_SLAVE, NULL) = 0
> ```

We probably still need a comment that while we made this MS_SLAVE here, that's ok cause we aren't doing anything else below.